pymongo/__init__.py

```diff
@@ -87,11 +88,13 @@
 
 from pymongo import _csot
 from pymongo._version import __version__, get_version_string, version_tuple
-from pymongo.collection import ReturnDocument
-from pymongo.common import MAX_SUPPORTED_WIRE_VERSION, MIN_SUPPORTED_WIRE_VERSION
-from pymongo.cursor import CursorType
-from pymongo.mongo_client import MongoClient
-from pymongo.operations import (
+from pymongo.cursor_shared import CursorType
+
+# isort: off
+from pymongo.synchronous.collection import ReturnDocument
+from pymongo.synchronous.common import MAX_SUPPORTED_WIRE_VERSION, MIN_SUPPORTED_WIRE_VERSION
+from pymongo.synchronous.mongo_client import MongoClient
+from pymongo.synchronous.operations import (
```

Comment:
Why do we have multiple operations classes? IMO there should only be one set of InsertOne, etc... operations which can be passed to either sync or async apis.

Reply:
This is a consequence of the mixing of data and IO operations in much of our codebase.  I completely agree we should only have a single set of these operations classes.

Reply:
We could decide to do that refactoring and further separation now, but that would take a significant amount of time and delay this beta release. We'd also continue to have the constant merge conflicts and rebasing issues on this PR as more commits are merged to master.

Reply:
I agree, [PYTHON-4476](https://jira.mongodb.org/browse/PYTHON-4476) is tracking this.

Reply:
Okay as long as [PYTHON-4476](https://jira.mongodb.org/browse/PYTHON-4476) is implemented before we release anything non-alpha (betas aren't intended to have breaking changes either).